edge/pkg/edgehub/clients/wsclient/websocket.go

```diff
@@ -105,13 +105,22 @@ func (wsc *WebSocketClient) UnInit() {
 
 //Send sends the message as JSON object through the connection
 func (wsc *WebSocketClient) Send(message model.Message) error {
+	err := wsc.connection.SetWriteDeadline(time.Now().Add(wsc.config.WriteDeadline))
+	if err != nil {
+		return err
+	}
 	return wsc.connection.WriteMessageAsync(&message)
 }
 
 //Receive reads the binary message through the connection
 func (wsc *WebSocketClient) Receive() (model.Message, error) {
+	var err error
 	message := model.Message{}
-	err := wsc.connection.ReadMessage(&message)
```

Comment:
Looks good, but seems `ReadMessage` hadn't use this Deadline in  Underlying `WSConnection`?

Reply:
Writing use it here: https://github.com/kubeedge/kubeedge/blob/master/staging/src/github.com/kubeedge/viaduct/pkg/conn/ws.go#L179

Reply:
> Looks good, but seems `ReadMessage` hadn't used this Deadline in Underlying `WSConnection`?

Yes. That's my mistake. `ReadMessage` gets messages from messageFifo in `WSConnection`, the read deadline setting also does't take effect in this method.
But CloudHub won't regularly send messages to the edge to make sure the websocket connection is alive. So is it necessary to set a timeout when read message?

Reply:
delete the code for setting read message deadline

Reply:
ok, I'm also not sure, maybe we don't need set read deadline?